### PR TITLE
[Bug] Fix CI linter errors

### PR DESCRIPTION
**Pull request by Olshansk**
### Summary generated by Reviewpad on 10 Jul 23 23:40 UTC
This pull request includes the following changes:

- Added a new file `.github/workflows/golangci-lint.yml` for golangci-lint workflow.
- Modified the `.github/workflows/main.yml` file by removing the `golangci-lint` job.
- Updated the `.golangci.yml` file to remove the `typecheck` linter.
- Updated the `Makefile` to add installation of `helm` and `tilt` as part of `install_cli_deps` task.
- Modified the `app/client/cli/consensus.go` file by adding an import statement.
- Modified the `p2p/config/config.go` file to return an error in the `IsValid()` method.
- Modified the `p2p/raintree/router.go` file by removing the unused import and variable.

Overall, this pull request adds a new golangci-lint workflow, updates some workflow files, makes changes to the Makefile, and modifies a few Go files.

## Origin Document

I merged something unrelated to main and saw that the CI was broken

https://github.com/pokt-network/pocket/actions/runs/5468898070

Screenshot 2023-07-05 at 3 46 15 PM

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Fix linter errors on main
- Make the linter run on branches as a separate workflow

## Testing

- [ ] `make develop_test`; if any code changes were made
- [ ] `make test_e2e` on [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any code changes were made
- [ ] `e2e-devnet-test` passes tests on [DevNet](https://pocketnetwork.notion.site/How-to-DevNet-ff1598f27efe44c09f34e2aa0051f0dd); if any code was changed
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [ ] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

### CODE REVIEW

**Comment by okdas** [comment]
Thanks @Olshansk! I opened a PR (#850) to fix this and had planned to return to it. However, I like your concept of employing a separate `golangci-lint` workflow. 

The existing approach configures linter to run only against modified files, allowing some errors to slip through. If we constantly run the linter on the entire codebase (by setting `only-new-issues` to false), it'll be more effective. It also won't impact the CI times, as it runs in parallel without delaying the main build.

To avoid redundancy, let's remove the [current](https://github.com/pokt-network/pocket/blob/linter_fixes/.github/workflows/main.yml#L90) linter step from the main pipeline and get rid of `only-new-issues: true`. 

I can apply these changes to your branch if that's okay.

**Comment by Olshansk** [comment]
> I can apply these changes to your branch if that's okay.

Sgtm, go ahead!

**Comment by okdas** [comment]
@Olshansk with linter now checking the whole codebase, it found more issues. I'll investigate them tomorrow.

**Comment by Olshansk** [comment]
> @Olshansk with linter now checking the whole codebase, it found more issues. I'll investigate them tomorrow.

@okdas Have you had a chance to look into this?

**Comment by okdas** [comment]
> @okdas Have you had a chance to look into this?

Yes, I'm working on this now. I wasn't able to reproduce locally, though, and have some ideas as of why this is happening.

**Comment by okdas** [comment]
@Olshansk Here's what's happening. The new CI job only pulls the source code, which doesn't include any code-generated stuff. As a result, we got a bunch of `typecheck` errors. I've added the necessary dependencies for that CI job, and it now comes back green.

I tried to remove the `typecheck` linter because it is redundant anyway (it does the same check the Go compiler performs during compilation), but there is no way to disable it via config, and some CLI voodoo is required.

At this point, I think it should be good to go (GTG). I've approved the PR; please merge if you are okay with my changes.

**Review by okdas — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -121,12 +121,16 @@ go_fmt: ## Format all the .go files in the project in place.
 
 .PHONY: install_cli_deps
 install_cli_deps: ## Installs `protoc-gen-go`, `mockgen`, 'protoc-go-inject-tag' and other tooling
```
Comment (Olshansk) [inline Makefile:123]:
```suggestion
install_cli_deps: ## Installs `helm`, `tilt` and the underlying `ci_deps`
```

File: Makefile [hunk 2]
```diff
@@ -121,12 +121,16 @@ go_fmt: ## Format all the .go files in the project in place.
...
+	make install_ci_deps
+	curl -fsSL https://raw.githubusercontent.com/tilt-dev/tilt/master/scripts/install.sh | bash
+	curl -fsSL https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
+
+.PHONY: install_ci_deps
+install_ci_deps: ## Installs `protoc-gen-go`, `mockgen`, 'protoc-go-inject-tag' necessary for CI
```
Comment (Olshansk) [inline Makefile:129]:
```suggestion
install_ci_deps: ## Installs `protoc-gen-go`, `mockgen`, 'protoc-go-inject-tag' and other tools necessary for CI
```

File: .golangci.yml [hunk 3]
```diff
@@ -1,24 +1,23 @@
 linters:
   # Full list available at https://golangci-lint.run/usage/linters/#disabled-by-default
+  disable-all: true
```
Comment (Olshansk) [inline .golangci.yml:3]: @okdas Why keep the linters here if it's disabled?

1. We should delete it
OR
2. Add the comment below

Wanted to understand why not do (1)

```suggestion
  disable-all: true # Disabled and kept around as a reference. See # golangci-lint.yml instead
```

Comment (okdas) [reply .golangci.yml:3]: @Olshansk, it doesn't matter if we keep it or remove it. We list the linters we enable, even the ones enabled by default. I introduced it to try to remove the `typecheck` linter, but it seems it's not possible without CLI voodoo. Let's remove it if it causes any confusion.

Comment (Olshansk) [reply .golangci.yml:3]: > Let's remove it if it causes any confusion.

Correct. Dead code -> raises questions -> increases cognitive load -> confuses developers -> leads to time spent on irrelevant things -> reduces productivity

**Pull request merged**